x/wasm/types/types.go

```diff
@@ -83,6 +90,60 @@ func (c *ContractInfo) ValidateBasic() error {
 	if err := validateLabel(c.Label); err != nil {
 		return sdkerrors.Wrap(err, "label")
 	}
+	if c.Extension == nil {
+		return nil
+	}
+	e, ok := c.Extension.GetCachedValue().(validatable)
+	if !ok {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "extension: %T", c.Extension)
+	}
+	if err := e.ValidateBasic(); err != nil {
+		return sdkerrors.Wrap(err, "extension")
+	}
+	return nil
+}
+
+// SetExtension set new extension data
+func (c *ContractInfo) SetExtension(ext ContractInfoExtension) error {
+	if e, ok := ext.(validatable); ok {
```

Comment:
But here, you allow them to set one that doesn't implement `ValidateBasic` (treating that as "always valid").

Please unify this with the above function and document - either `ValidateBasic` is optional, and it is always valid if missing, or `ValidateBasic` is required and it is always invalid if missing. In any case, `ValidateBasic` is run on the extension when setting it and when validating ContractInfo